flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/data/util/DataFormatConverters.java

```diff
@@ -1138,7 +1138,7 @@ private ArrayData toBinaryArray(T[] value) {
 				}
 			}
 			reuseWriter.complete();
-			return reuseArray;
+			return reuseArray.copy();
```

Comment:
@zoudan Thanks for your contribution, could you also add a test to cover this?

Reply:
OK，I will add a test for this.

Reply:
Add tests for this.